src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static add(creature) {
+    this.alive.push(creature);
+  }
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.add(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(creature) {
+    if (creature instanceof Herbivore && !creature.hidden) {
+      creature.health -= 50;
+    }
+
+    if (creature.health === 0) {
+      Animal.alive = Animal.alive.filter(beast => beast.health > 0);
```

Comment:
you could use filter without condition

Reply:
I could, but I think it will lead to unnecessary iterations over the array. Every time the bite method is called, and the first condition is satisfied, the filter method will be executed.